garage/tf/algos/ddpg.py

```diff
@@ -20,50 +11,78 @@
 
 
 class DDPG(OffPolicyRLAlgorithm):
-    """
-    A DDPG model based on https://arxiv.org/pdf/1509.02971.pdf.
+    """A DDPG model based on https://arxiv.org/pdf/1509.02971.pdf.
+
+    DDPG, also known as Deep Deterministic Policy Gradient, uses actor-critic
+    method to optimize the policy and reward prediction. It uses a supervised
+    method to update the critic network and policy gradient to update the actor
+    network. And there are exploration strategy, replay buffer and target
+    networks involved to stabilize the training process.
 
     Example:
         $ python garage/examples/tf/ddpg_pendulum.py
     """
 
     def __init__(self,
                  env_spec,
+                 policy,
+                 qf,
                  replay_buffer,
+                 n_epoch_cycles=20,
+                 n_train_steps=50,
+                 max_path_length=None,
+                 buffer_batch_size=64,
+                 min_buffer_size=int(1e4),
+                 rollout_batch_size=1,
+                 exploration_strategy=None,
                  target_update_tau=0.01,
                  policy_lr=1e-4,
                  qf_lr=1e-3,
+                 discount=0.99,
                  policy_weight_decay=0,
                  qf_weight_decay=0,
                  policy_optimizer=tf.train.AdamOptimizer,
                  qf_optimizer=tf.train.AdamOptimizer,
                  clip_pos_returns=False,
                  clip_return=np.inf,
-                 discount=0.99,
                  max_action=None,
-                 name=None,
-                 **kwargs):
+                 reward_scale=1.,
+                 input_include_goal=False,
+                 smooth_return=True,
+                 name='DDPG'):
         """
-        Construct class.
-
         Args:
-            env_spec(EnvSpec): Environment specification.
-            target_update_tau(float): Interpolation parameter for doing the
-        soft target update.
+            env_spec (EnvSpec): Environment specification.
+            policy (garage.tf.policies.base.Policy): Policy.
```

Comment:
please document constructor arguments in the class docstring, as mentioned in `CONTRIBUTING.md`